yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/compiler/code/ExactInvocation.java

```diff
@@ -4,25 +4,42 @@
  * See LICENSE file for terms.
  */
 
-package com.yahoo.yqlplus.engine.internal.bytecode.types.gambit;
+package com.yahoo.yqlplus.engine.compiler.code;
 
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
-import com.yahoo.yqlplus.engine.internal.compiler.CodeEmitter;
-import com.yahoo.yqlplus.engine.internal.plan.types.BytecodeExpression;
-import com.yahoo.yqlplus.engine.internal.plan.types.TypeWidget;
+import com.google.inject.TypeLiteral;
 import com.yahoo.yqlplus.language.parser.Location;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 import java.util.Arrays;
 import java.util.List;
 
 public class ExactInvocation extends BytecodeInvocable {
+    public static GambitCreator.Invocable reflectInvoke(EngineValueTypeAdapter adapter, TypeLiteral<?> owner, Method method) {
+        TypeWidget ownerType = adapter.adapt(owner);
+        TypeWidget returnType = adapter.adapt(owner.getReturnType(method));
+        String methodName = method.getName();
+        List<TypeWidget> argTypes = Lists.newArrayListWithExpectedSize(method.getParameterCount());
+        for(TypeLiteral<?> argType : owner.getParameterTypes(method)) {
+            argTypes.add(adapter.adapt(argType));
+        }
+        int op = Opcodes.INVOKEVIRTUAL;
+        if(owner.getRawType().isInterface()) {
+            op = Opcodes.INVOKEINTERFACE;
+        }
+        if(Modifier.isStatic(method.getModifiers())) {
```

Comment:
missing space

Reply:
Pushed an update that ran Code Format on all files -- there were many missing spaces even beyond all these.